apps/api/src/domain/models/song/commands/add-song.command.ts

```diff
@@ -0,0 +1,37 @@
+import { Command, ICommand } from '@coscrad/commands';
+import {
+    IsNonNegativeFiniteNumber,
+    IsOptional,
+    IsStringWithNonzeroLength,
+    IsUrl,
+    IsUUID,
+    ValidateNested,
+} from '@coscrad/validation';
+import { ContributorAndRole } from '../ContributorAndRole';
+
+@Command('ADD_SONG')
+export class AddSong implements ICommand {
+    @IsUUID()
```

Comment:
So we're implementing UUID?

Reply:
I had hoped this would spark a conversation. I recall you had lots of good arguments in favour of this.

There is a conceptual issue though. If the front-end sends us the UUID, we can logicaly guarantee they actually generated a new ID (we can only guarantee the format). There's no efficient way to validate that a UUID is unique across our system, except to generate it honestly on the back-end.

At the same time, there are many cases where the client knowing the UUID and leveraging this in a subsequent action (e.g. fetch updated state upon success acknowledgement from the command, bulk upload multiple models linked by reference) is convenient. We should talk more about this before we finalize the command module.